yt_dlp/extractor/taptap.py

```diff
@@ -0,0 +1,209 @@
+        }]
+    }]
+
+
+class TapTapAppIE(TapTapBaseIE):
+    _VALID_URL = r'https?://www\.taptap\.cn/app/(?P<id>\d+)'
+    _INFO_API = 'https://www.taptap.cn/webapiv2/app/v4/detail'
+    _ID_PATH = (('app_videos', 'videos'), ..., 'video_id')
+    _META_PATH = {
+        'title': ('title', {str}),
+        'description': ('description', 'text', {str}, {clean_html}),
+    }
+    _TESTS = [{
+        'url': 'https://www.taptap.cn/app/168332',
+        'info_dict': {
+            'id': '168332',
+            'title': '原神',
+            'description': 'md5:e345f39a5fea5de2a46923f70d5f76ab',
+        },
+        'playlist_count': 2,
+        'playlist': [{
+            'info_dict': {
+                'id': '4058443',
+                'ext': 'mp4',
+                'title': '原神',
+                'description': 'md5:e345f39a5fea5de2a46923f70d5f76ab',
+                'duration': 26,
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }, {
+            'info_dict': {
+                'id': '4058462',
+                'ext': 'mp4',
+                'title': '原神',
+                'description': 'md5:e345f39a5fea5de2a46923f70d5f76ab',
+                'duration': 295,
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }]
+    }]
+
+
+class TapTapAppIntlIE(TapTapAppIE, TapTapIntlBase):
+    _VALID_URL = r'https?://www\.taptap\.io/app/(?P<id>\d+)'
+    _INFO_API = 'https://www.taptap.io/webapiv2/i/app/v5/detail'
+    _DATA_PATH = ('data', 'app')
+
+
+class TapTapPostIntlIE(TapTapAppIntlIE):
+    _VALID_URL = r'https?://www\.taptap\.io/post/(?P<id>\d+)'
+    _INFO_API = 'https://www.taptap.io/webapiv2/creation/post/v1/detail'
+    _INFO_QUERY_KEY = 'id_str'
+    _DATA_PATH = ('data', 'post')
+    _ID_PATH = ((('videos', ...), 'pin_video'), 'video_id')
+    _META_PATH = {
+        'timestamp': ('published_time', {int_or_none}),
+        'uploader': ('user', 'name', {str}),
+        'title': ('title', {str}),
+        'description': ('list_fields', 'summary', {str}),
+    }
+    _TESTS = [{
+        'url': 'https://www.taptap.io/post/571785',
+        'info_dict': {
+            'id': '571785',
+            'title': 'Arknights x Rainbow Six Siege | Event PV',
+            'description': 'md5:f7717c13f6d3108e22db7303e6690bf7',
+            'timestamp': 1614664951,
+            'upload_date': '20210302',
+            'uploader': 'TapTap Editor',
+        },
+        'playlist_count': 1,
+        'playlist': [{
+            'info_dict': {
+                'id': '2149491903',
+                'ext': 'mp4',
+                'title': 'Arknights x Rainbow Six Siege | Event PV',
+                'description': 'md5:f7717c13f6d3108e22db7303e6690bf7',
+                'duration': 122,
+                'timestamp': 1614664951,
+                'upload_date': '20210302',
+                'uploader': 'TapTap Editor',
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }]
```

Comment:
```suggestion
        }],
        'params': {'skip_download': 'm3u8'},
```

Reply:
I seems that only h265 format somehow fails test (perhaps a future tester should fix this?).
I wonder if replacing `'skip_download'` with `'format_sort': ['vcodec:h264']` is better since it tests h264 download at least.

Reply:
I think this happens due to the size of the first segment in the m3u8 playlist(?) Not really sure, though.
If you grep for `'skip_download':` you will find lots of tests that skip downloading. I think it's fine to skip the download test.

Reply:
I see. With `test` option, downloader only get the first fragment. I wonder if it is worth changing downloaders to download first 2 or 3 fragments, or provide a param to tweak it: https://github.com/yt-dlp/yt-dlp/blob/78c57cc0e0998b8ed90e4306f410aa4be4115cd7/yt_dlp/downloader/hls.py#L270-L271
```diff
- fragments = [fragments[0] if fragments else None]
+ fragments = fragments[:self.params.get('test_download_frags', 2)] if fragments else None
```